File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int t) {
        vector<int> ans;
        long long int n = arr.size();
        unordered_map<int,int> mp;
        for(long long int i = 0;i<n;i++){
            if(mp.find(t-arr[i]) != mp.end()){
                long long temp = t-arr[i];
                ans.emplace_back(mp[temp]);
                ans.emplace_back(i);
                return ans;
            }
            else mp[arr[i]] = i;
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] arr, int t) {
        Dictionary<int, int> dict = new();

        int[] res = new int[2];
        for(int i = 0; i < arr.Length; i++){
            if(dict.ContainsKey(arr[i])){
                res[0] = dict[arr[i]];
                res[1] = i;
                return res;
            }
            else{
                dict[t - arr[i]] = i;
            }
        }
        return new int[2];
    }
}